zomg.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+// ruleid: checklist-cookie-fetch
+# Do not use $_COOKIE to get cookie values
+echo 'Hello ' . htmlspecialchars($_COOKIE["name"]) . '!';
+$email = $_COOKIE["email"];
```

Comment:
Directly accessing of $_COOKIE is not considered best practice, please use `$wgRequest->getCookie(...)` instead. ([php_cookie_autofix](https://semgrep.dev/r/None))

You can apply this patch to fix the issue:

```suggestion
$email = $wgRequest->getCookie("email")
;
```



#
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text_horizontal.svg" height=16px alt="" />
</a>

<sup>Please address this comment before merging this pull request.</sup>
<sup>[How do I suppress this?](https://semgrep.dev/docs/ignoring-findings/) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=php_cookie_autofix) **·** [Manage rule](https://semgrep.dev/manage/projects)</sup>